src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -1189,76 +1251,135 @@ private static int GetDeclarationDepth(SyntaxTrivia trivia)
+                {
+                    // Nested statements are normally indented one level.
+                    //
+                    // However, for chains of using-statements or fixed-statements, we'd like to follow the
+                    // idiomatic pattern of:
+                    //
+                    //      using ...
+                    //      using ...
+                    //          .. embedded statement ..
+                    if (node is UsingStatementSyntax { Parent: UsingStatementSyntax })
+                        return parentDepth;
 
-                        return parentDepth + 1;
-                    }
+                    if (node is FixedStatementSyntax { Parent: FixedStatementSyntax })
+                        return parentDepth;
 
-                    if (node is MemberDeclarationSyntax ||
-                        node is AccessorDeclarationSyntax ||
-                        node is TypeParameterConstraintClauseSyntax ||
-                        node is SwitchSectionSyntax ||
-                        node is SwitchExpressionArmSyntax ||
-                        node is UsingDirectiveSyntax ||
-                        node is ExternAliasDirectiveSyntax ||
-                        node is QueryExpressionSyntax ||
-                        node is QueryContinuationSyntax)
-                    {
-                        return parentDepth + 1;
-                    }
+                    return parentDepth + 1;
+                }
 
-                    return parentDepth;
+                if (node is MemberDeclarationSyntax ||
+                    node is AccessorDeclarationSyntax ||
+                    node is TypeParameterConstraintClauseSyntax ||
+                    node is SwitchSectionSyntax ||
+                    node is SwitchExpressionArmSyntax ||
+                    node is UsingDirectiveSyntax ||
+                    node is ExternAliasDirectiveSyntax ||
+                    node is QueryExpressionSyntax ||
+                    node is QueryContinuationSyntax)
+                {
+                    return parentDepth + 1;
                 }
+
+                return parentDepth;
             }
 
             return 0;
         }
 
+        /// <summary>
+        /// Tells if the given SyntaxNode is inside single-line initializer context.
+        /// Initializers in such context are not expected to be large,
+        /// so formatting them in single-line fashion looks more compact.
+        /// Current cases:
+        /// <list type="bullet">
+        /// <item>Interpolation holes in strings</item>
+        /// <item>Attribute arguments</item>
+        /// <item>Normal arguments</item>
+        /// </list>
+        /// </summary>
+        private static bool IsSingleLineInitializerContext(SyntaxNode? node)
+        {
+            if (node is null)
+            {
+                return false;
+            }
+
+            var currentParent = node.Parent;
+
+            while (currentParent is not null)
```

Comment:
nit: we can also bail out early if we reach a statement (`StatementSyntax`) or member declaration (`MemberDeclarationSyntax`, for scenario like `: this(...)` constructor initializer).

Reply:
I don't agree with statements. Consider the following cases:
```cs
var a = new A
{
  A = 0
};
```
```cs
if (true)
{
  new A
  {
    A = 0
  };
}
```
```cs
try
{
  new A
  {
    A = 0
  };
}
catch
{
}
```
Here on of the parents is a `StatementSyntax` (`LocalStatementSyntax`, `IfStatementSyntax`, `TryStatementSyntax`) but the initializer should be multiline. Can clarify cases where you want to see singleline initializer?

Reply:
Well, added a few test for statements and searching for `MemberDeclarationSyntax` is not true either, consider this:
```cs
class C
{
  void M()
  {
    new SomeClass
    {
      A = 1,
      B = 2,
      C = 3
    };
  }
}
```
So I removed this check as well. Added a test for `: base(...)` (which is equvivalent to `: this(...)`) case which is already covered by checking for `ArgumentSyntax`

Reply:
A statement or member declaration can never be inside an interpolation or attribute or argument. We're looking to see if we're in one of those three.
So if we reach a statement or member declaration we can return false without further exploration.

Reply:
Uh, got it. Done!